port/main.c

```diff
@@ -49,6 +50,13 @@
 	} while (0)
 
 
+#if LWIP_IPSEC
+static int ipsecdev_disconnected;
```

Comment:
I believe our coding convention is to put all global variables to one "common" struct.

I like the simplicity of this solution, but the naming is misleading (link monitoring is not connected to IPSec in any way, just the interface name is taken from `LWIP_IPSEC_DEV`).

I think we could either:
 - provide separate define for monitoring device (so the functionality would not be under `LWIP_IPSEC`)
 - add support for writing device name in runtime (so to start monitoring the device I would do `echo -n "en1" > /dev/linkmonitor`) and than start reading the file?